### ISSUE TITLE
Can't search for nodes by type where the type name contains a space

### ISSUE THREAD

**Issue by rmwiseman** [node-red/node-red#4071]
### Current Behavior

If I want to find `link out` nodes (for example), I can go to the palette and click the magnifying glass icon:

![image](https://user-images.githubusercontent.com/10866705/221201621-a0d7ae33-f1b7-4e77-8902-b70ad83a1a1e.png)

This works for nodes without spaces in the name, but for things like `link out`, nothing is found.  I've tried modifying the search from:

`type:link out`

to:

`"type:link out"` or `type:"link out"`

but nothing matches.

### Expected Behavior

I would expect all `link out` nodes (or whatever was searched for) to be listed.

### Steps To Reproduce

In a flow containing nodes with spaces in the name, hover over the node type in the palette and choose its magnifying glass icon (as shown in the picture above).  Nothing will match the search.

### Example flow

```
[
    {
        "id": "f9d9669f9b22e9d5",
        "type": "tab",
        "label": "Node type search test",
        "disabled": false,
        "info": "",
        "env": []
    },
    {
        "id": "bcfda855ca09f51a",
        "type": "inject",
        "z": "f9d9669f9b22e9d5",
        "name": "\"inject\" node",
        "props": [
            {
                "p": "payload"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "",
        "payload": "true",
        "payloadType": "bool",
        "x": 170,
        "y": 120,
        "wires": [
            []
        ]
    },
    {
        "id": "3e888eb0a49c035c",
        "type": "comment",
        "z": "f9d9669f9b22e9d5",
        "name": "1. Search for nodes of type \"inject\" by selecting the magnifying glass icon from the node in the palette.\\n The node below will be found.",
        "info": "",
        "x": 430,
        "y": 60,
        "wires": []
    },
    {
        "id": "565ce0c51414fe72",
        "type": "http in",
        "z": "f9d9669f9b22e9d5",
        "name": "\"http in\" node",
        "url": "/something",
        "method": "get",
        "upload": false,
        "swaggerDoc": "",
        "x": 150,
        "y": 260,
        "wires": [
            []
        ]
    },
    {
        "id": "cb4ae674ba78a8cf",
        "type": "comment",
        "z": "f9d9669f9b22e9d5",
        "name": "2. Search for nodes of type \"http in\" by selecting the magnifying glass icon from the node in the palette.\\n The node below will NOT be found.  Indeed, no nodes will be found.",
        "info": "",
        "x": 430,
        "y": 200,
        "wires": []
    }
]
```

### Environment

- Node-RED version: 3.0.2
- Node.js version: 19.0.0
- npm version: 9.2.0
- Platform/OS: Mint 21.1 (based upon Ubuntu 22.04)
- Browser: Firefox 110.0

**Comment by Steve-Mcl** [comment]
this is resolved in beta 3.1.0

**Comment by knolleary** [comment]
Resolved via #3841
